File: is_prime/is_prime.cpp
#include <iostream>
using namespace std;

bool is_prime(int n) {
	if (n == 2 || n == 3)
		return true;
	if (n % 2 == 0 || n % 3 == 0)
		return false;
	for (int i = 1; 36*i*i-12*i < n ; i++)
		if ((n % (6*i+1) == 0) || (n % (6*i-1) == 0)) 
			return false;
	return true;
}

File: is_prime/is_prime.ts
/**
 * To run this file:
 * In the project root run:
 * 1. npm install
 * 2.  npm run tsc is_prime\is_prime.ts
 */

const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Please enter a number: ', (numberString: string) => {
    try {
        const number: number = Number(numberString);
        console.log(`The number: ${number} is ${isPrime(number) ? '':'not '} prime.`);
    } catch (e) {
        console.log(`The ${numberString} is not a number: ${e}`)
    }
    rl.close();
});

function isPrime(number: number): boolean {
    let i = 0;
    for(let j = 1; j <= number; j++) {
        if(number % j === 0) i = i + 1;
    }
    return i === 2;
}

